crates/core/src/subscription/query.rs

```diff
@@ -544,6 +546,12 @@ mod tests {
     #[test]
     fn test_eval_incr_for_index_join() -> ResultTest<()> {
         let (db, _) = make_test_db()?;
+        run_eval_incr_for_index_join(&db)?;
+        run_eval_incr_for_index_join(&db.with_row_count(Arc::new(|_| 5)))?;
```

Comment:
This new test case will run our incremental evaluation suite under the new query plan.